tools/tpm2_ecdhkeygen.c

```diff
@@ -92,13 +127,24 @@ static tool_rc process_inputs(ESYS_CONTEXT *ectx) {
      * 4. Configuration for calculating the pHash
      */
 
+
     /*
      * 4.a Determine pHash length and alg
      */
+    tpm2_session *all_sessions[MAX_SESSIONS] = {
+        ctx.session,
+        0,
+        0
+    };
+
+        const char **cphash_path = ctx.cp_hash_path ? &ctx.cp_hash_path : 0;
 
+    ctx.parameter_hash_algorithm = tpm2_util_calculate_phash_algorithm(ectx,
+        cphash_path, &ctx.cp_hash, 0, 0, all_sessions);
```

Comment:
Deriving pHash algorithm from input params is a separate commit, but this is ok.